nomad/structs/config/consul.go

```diff
@@ -98,8 +98,8 @@ func (a *ConsulConfig) Merge(b *ConsulConfig) *ConsulConfig {
 	if b.ClientServiceName != "" {
 		result.ClientServiceName = b.ClientServiceName
 	}
-	if b.AutoAdvertise {
-		result.AutoAdvertise = true
+	if !b.AutoAdvertise {
```

Comment:
All the booleans in the config really need to be updated to this style: https://github.com/hashicorp/nomad/blob/master/nomad/structs/config/vault.go#L26

The difference is that by making it a pointer we can tell if the user set it. This gives us three states: unset, false, true. If it is Unset we use the default and if not use what ever is set.

https://github.com/hashicorp/nomad/blob/master/nomad/structs/config/vault.go#L132